### ISSUE TITLE
feature: get last errorcode/errorlevel

### ISSUE THREAD

**Issue by bw1faeh0** [chrisant996/clink#128]
I tried it with the help of lua, but `os.getenv('errorlevel')` always returns nil, even if `echo %errorlevel%` in a cmd returns values.

Next step I tried something like this:
````
local function get_last_errorcode()
    local f = assert(io.popen("echo %ERRORLEVEL%", 'r'))
    local s = assert(f:read('*a'))
    f:close()
    s = string.gsub(s, '^%s+', '')
    s = string.gsub(s, '%s+$', '')
    s = string.gsub(s, '[\n\r]+', ' ')
    return s
end
````
But unfortunatly I always get a `0` even I'm sure that %ERRORLEVEL% is != 0.

Is there a way that clink can provide the errorlevel/errorcode of the last command that was executed?

**Comment by chrisant996** [comment]
This is a duplicate of #40 ([this comment](https://github.com/chrisant996/clink/issues/40#issuecomment-751522118)) and #91.

Using `io.popen()` starts a _new_ cmd.exe, which has its own separate %ERRORLEVEL% variable.

See the links above for more info.

**Comment by chrisant996** [comment]
Ok, I don't know how reliably this will work in practice, but I've hooked up an approach that's able to retrieve the value.  But it's a bit invasive, so it's off by default.

A new `cmd.get_errorlevel` setting can make Clink run a hidden `echo %errorlevel%` command before every interactive prompt and retrieve the last exit code value for use by Lua scripts via a new `os.geterrorlevel()` function (and `os.getenv("errorlevel")` also returns it).  It will be included in Clink v1.2.14 which will be released sometime in the next week or two.
